### ISSUE TITLE
Compatibility with slurm 19.05.01

### ISSUE THREAD

**Issue by dpryan79** [natefoo/slurm-drmaa#28]
At least when I built it just now, slurm 19.05.01 doesn't ship a libslurmdb.so, so the linking test run by configure fails. Removing the linking to it here solves the issue https://github.com/natefoo/slurm-drmaa/blob/master/m4/ax_slurm.m4#L75 but it'd be good if someone else confirmed before this is implemented.

**Comment by EricR86** [comment]
I can confirm that the slurmdb is merged in with the slurm library. There is a note from Schedmd:
```
NOTE: libslurmdb has been merged into libslurm.  If functionality is needed
      from libslurmdb please just link to libslurm.
```

It is under the "28 May 2019" section of the `RELEASE_NOTES` and can be found on [the website](https://slurm.schedmd.com/news.html)

**Comment by ocfmatt** [comment]
I have a compilation issue whereby I encounter the following error:
```
checking for usable SLURM libraries/headers... *** The SLURM test program failed to link or run. See the file config.log
*** for the exact error that occured.
no
configure: error:
Slurm libraries/headers not found;
add --with-slurm-inc and --with-slurm-lib with appropriate locations.
```

My configure command is:
`./configure --prefix=/opt/software/galaxy/plugins/drmaa --with-slurm-inc=/opt/software/slurm/19.05.1-2/include --with-slurm-lib=/opt/software/slurm/19.05.1-2/lib`

Libraries and includes are present as stated in the help text:
```
[root@maxlogin1 slurm-drmaa]# ls -la /opt/software/slurm/19.05.1-2/include/slurm/slurm.h /opt/software/slurm/19.05.1-2/lib/libslurm.a
-rw-r--r--. 1 root root   217039 Aug  5 11:24 /opt/software/slurm/19.05.1-2/include/slurm/slurm.h
-rw-r--r--. 1 root root 55722378 Aug  5 11:22 /opt/software/slurm/19.05.1-2/lib/libslurm.a
```

My Slurm installation is compiled from source using Intel compilers. I am getting the same error when running against Slurm 18.08.8 making me think I am potentially missing a compilation flag in either Slurm or drmaa.

@dpryan79 did you install Slurm 19.05.1 from packages or compile from source?

**Comment by EricR86** [comment]
@ocfmatt if you look in your `config.log` I'm pretty certain the error will show up as 
`/usr/bin/ld: cannot find -lslurmdb`
after configure tries to compile a test program. It's complaining specifically that it can't find libslurmdb.so because that file no longer exists in Slurm 19.05.

You can remove the `-lslurmdb` line safely in either the fix shown above or from the `SLURM_LIBS` variable inside `configure` itself (it was around line 14034 for me).

**Comment by ocfmatt** [comment]
@EricR86 Thanks for your quick reply.

I removed "-lslurmdb" and was able to configure and make after I downloaded the release source code. I had an uninitialised drmaa_utils path causing other errors.

**Comment by research-computing-facility** [comment]
I have been able to build the drmaa thanks to these tips with slurm 19.04 but the module will not load:
`galaxy.jobs.runners.drmaa INFO 2019-11-21 17:13:08,365 Overriding DRMAA_LIBRARY_PATH due to runner plugin parameter: /galaxy/slurm-drmaa/compiled/lib/libdrmaa.so
Traceback (most recent call last):
  File "/galaxy/production/lib/galaxy/webapps/galaxy/buildapp.py", line 58, in paste_app_factory
    app = galaxy.app.UniverseApplication(global_conf=global_conf, **kwargs)
  File "/galaxy/production/lib/galaxy/app.py", line 189, in __init__
    self.job_manager = manager.JobManager(self)
  File "/galaxy/production/lib/galaxy/jobs/manager.py", line 24, in __init__
    self.job_handler = handler.JobHandler(app)
  File "/galaxy/production/lib/galaxy/jobs/handler.py", line 34, in __init__
    self.dispatcher = DefaultJobDispatcher(app)
  File "/galaxy/production/lib/galaxy/jobs/handler.py", line 779, in __init__
    self.job_runners = self.app.job_config.get_job_runner_plugins(self.app.config.server_name)
  File "/galaxy/production/lib/galaxy/jobs/__init__.py", line 649, in get_job_runner_plugins
    rval[id] = runner_class(self.app, runner['workers'], **runner.get('kwds', {}))
  File "/galaxy/production/lib/galaxy/jobs/runners/drmaa.py", line 63, in __init__
    drmaa = __import__("drmaa")
  File "/galaxy/production/.venv/lib/python2.7/site-packages/drmaa/__init__.py", line 65, in <module>
    from .session import JobInfo, JobTemplate, Session
  File "/galaxy/production/.venv/lib/python2.7/site-packages/drmaa/session.py", line 39, in <module>
    from drmaa.helpers import (adapt_rusage, Attribute, attribute_names_iterator,
  File "/galaxy/production/.venv/lib/python2.7/site-packages/drmaa/helpers.py", line 36, in <module>
    from drmaa.wrappers import (drmaa_attr_names_t, drmaa_attr_values_t,
  File "/galaxy/production/.venv/lib/python2.7/site-packages/drmaa/wrappers.py", line 56, in <module>
    _lib = CDLL(libpath, mode=RTLD_GLOBAL)
  File "/usr/lib64/python2.7/ctypes/__init__.py", line 360, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /galaxy/slurm-drmaa/compiled/lib/libdrmaa.so: undefined symbol: slurm_kill_job2`
Would anyone have any suggestions?
Many thanks in advance

**Comment by dpryan79** [comment]
libdrmaa should be linking against libslurm.so, but that doesn't seem to be the case for you, which I think is the cause of the problem.

**Comment by research-computing-facility** [comment]
@dpryan79 I believe you are right, as libslurm.so does not appear to be a library in the latest slurm:
```
[root@vmpr-res-cluster1 ~]# ls -l /usr/lib64/slurm/
total 31864
-rwxr-xr-x 1 root root  355240 Nov 18 03:19 accounting_storage_filetxt.so
-rwxr-xr-x 1 root root  240840 Nov 18 03:19 accounting_storage_none.so
-rwxr-xr-x 1 root root  451064 Nov 18 03:19 accounting_storage_slurmdbd.so
-rwxr-xr-x 1 root root  199048 Nov 18 03:19 acct_gather_energy_cray_aries.so
-rwxr-xr-x 1 root root  199096 Nov 18 03:19 acct_gather_energy_ibmaem.so
-rwxr-xr-x 1 root root  187328 Nov 18 03:19 acct_gather_energy_none.so
-rwxr-xr-x 1 root root  216592 Nov 18 03:19 acct_gather_energy_rapl.so
-rwxr-xr-x 1 root root  211816 Nov 18 03:19 acct_gather_filesystem_lustre.so
-rwxr-xr-x 1 root root  186800 Nov 18 03:19 acct_gather_filesystem_none.so
-rwxr-xr-x 1 root root  186824 Nov 18 03:19 acct_gather_interconnect_none.so
-rwxr-xr-x 1 root root  239840 Nov 18 03:19 acct_gather_interconnect_ofed.so
-rwxr-xr-x 1 root root  294104 Nov 18 03:19 acct_gather_profile_hdf5.so
-rwxr-xr-x 1 root root  268168 Nov 18 03:19 acct_gather_profile_influxdb.so
-rwxr-xr-x 1 root root  205000 Nov 18 03:19 acct_gather_profile_none.so
-rwxr-xr-x 1 root root  214400 Nov 18 03:19 auth_munge.so
-rwxr-xr-x 1 root root  584240 Nov 18 03:19 burst_buffer_datawarp.so
-rwxr-xr-x 1 root root  298552 Nov 18 03:19 burst_buffer_generic.so
-rwxr-xr-x 1 root root  194480 Nov 18 03:19 checkpoint_none.so
-rwxr-xr-x 1 root root  204112 Nov 18 03:19 checkpoint_ompi.so
-rwxr-xr-x 1 root root  206712 Nov 18 03:19 cli_filter_none.so
-rwxr-xr-x 1 root root  186968 Nov 18 03:19 core_spec_cray_aries.so
-rwxr-xr-x 1 root root  185264 Nov 18 03:19 core_spec_none.so
-rwxr-xr-x 1 root root  170408 Nov 18 03:19 cred_munge.so
-rwxr-xr-x 1 root root  158872 Nov 18 03:19 cred_none.so
-rwxr-xr-x 1 root root  217808 Nov 18 03:19 ext_sensors_none.so
-rwxr-xr-x 1 root root  255440 Nov 18 03:19 ext_sensors_rrd.so
-rwxr-xr-x 1 root root  197888 Nov 18 03:19 gpu_generic.so
-rwxr-xr-x 1 root root  292752 Nov 18 03:19 gres_gpu.so
-rwxr-xr-x 1 root root  247728 Nov 18 03:19 gres_mic.so
-rwxr-xr-x 1 root root  295328 Nov 18 03:19 gres_mps.so
-rwxr-xr-x 1 root root  247728 Nov 18 03:19 gres_nic.so
-rwxr-xr-x 1 root root  349064 Nov 18 03:19 jobacct_gather_cgroup.so
-rwxr-xr-x 1 root root  276256 Nov 18 03:19 jobacct_gather_linux.so
-rwxr-xr-x 1 root root  204568 Nov 18 03:19 jobacct_gather_none.so
-rwxr-xr-x 1 root root  276848 Nov 18 03:19 jobcomp_elasticsearch.so
-rwxr-xr-x 1 root root  239968 Nov 18 03:19 jobcomp_filetxt.so
-rwxr-xr-x 1 root root  395480 Nov 18 03:19 jobcomp_mysql.so
-rwxr-xr-x 1 root root  180072 Nov 18 03:19 jobcomp_none.so
-rwxr-xr-x 1 root root  229640 Nov 18 03:19 jobcomp_script.so
-rwxr-xr-x 1 root root  221008 Nov 18 03:19 job_container_cncu.so
-rwxr-xr-x 1 root root  186872 Nov 18 03:19 job_container_none.so
-rwxr-xr-x 1 root root  202280 Nov 18 03:19 job_submit_all_partitions.so
-rwxr-xr-x 1 root root  202288 Nov 18 03:19 job_submit_cray_aries.so
-rwxr-xr-x 1 root root  313192 Nov 18 03:19 job_submit_lua.so
-rwxr-xr-x 1 root root  226632 Nov 18 03:19 job_submit_pbs.so
-rwxr-xr-x 1 root root  188504 Nov 18 03:19 job_submit_require_timelimit.so
-rwxr-xr-x 1 root root  216000 Nov 18 03:19 job_submit_throttle.so
-rwxr-xr-x 1 root root  290192 Nov 18 03:19 launch_slurm.so
-rwxr-xr-x 1 root root  136976 Nov 18 03:19 layouts_power_cpufreq.so
-rwxr-xr-x 1 root root  136080 Nov 18 03:19 layouts_power_default.so
-rwxr-xr-x 1 root root  141648 Nov 18 03:19 layouts_unit_default.so
-rwxr-xr-x 1 root root 8524408 Nov 18 03:19 libslurmfull.so
-rwxr-xr-x 1 root root  182624 Nov 18 03:19 mcs_account.so
-rwxr-xr-x 1 root root  194704 Nov 18 03:19 mcs_group.so
-rwxr-xr-x 1 root root  179240 Nov 18 03:19 mcs_none.so
-rwxr-xr-x 1 root root  181872 Nov 18 03:19 mcs_user.so
-rwxr-xr-x 1 root root  199192 Nov 18 03:19 mpi_none.so
-rwxr-xr-x 1 root root  199192 Nov 18 03:19 mpi_openmpi.so
-rwxr-xr-x 1 root root  634184 Nov 18 03:19 mpi_pmi2.so
-rwxr-xr-x 1 root root  377664 Nov 18 03:19 node_features_knl_cray.so
-rwxr-xr-x 1 root root  313824 Nov 18 03:19 node_features_knl_generic.so
-rwxr-xr-x 1 root root  312616 Nov 18 03:19 power_cray_aries.so
-rwxr-xr-x 1 root root  236608 Nov 18 03:19 power_none.so
-rwxr-xr-x 1 root root  182176 Nov 18 03:19 preempt_none.so
-rwxr-xr-x 1 root root  192152 Nov 18 03:19 preempt_partition_prio.so
-rwxr-xr-x 1 root root  192576 Nov 18 03:19 preempt_qos.so
-rwxr-xr-x 1 root root  191912 Nov 18 03:19 priority_basic.so
-rwxr-xr-x 1 root root  322960 Nov 18 03:19 priority_multifactor.so
-rwxr-xr-x 1 root root   15560 Nov 18 03:24 private-tmpdir.so
-rwxr-xr-x 1 root root  206928 Nov 18 03:19 proctrack_cgroup.so
-rwxr-xr-x 1 root root  208656 Nov 18 03:19 proctrack_linuxproc.so
-rwxr-xr-x 1 root root  186768 Nov 18 03:19 proctrack_pgid.so
-rwxr-xr-x 1 root root  186608 Nov 18 03:19 route_default.so
-rwxr-xr-x 1 root root  202240 Nov 18 03:19 route_topology.so
-rwxr-xr-x 1 root root  390824 Nov 18 03:19 sched_backfill.so
-rwxr-xr-x 1 root root  227144 Nov 18 03:19 sched_builtin.so
-rwxr-xr-x 1 root root  185424 Nov 18 03:19 sched_hold.so
-rwxr-xr-x 1 root root  602928 Nov 18 03:19 select_cons_res.so
-rwxr-xr-x 1 root root  695200 Nov 18 03:19 select_cons_tres.so
-rwxr-xr-x 1 root root  424712 Nov 18 03:19 select_cray_aries.so
-rwxr-xr-x 1 root root  400512 Nov 18 03:19 select_linear.so
-rwxr-xr-x 1 root root  210360 Nov 18 03:19 site_factor_none.so
-rwxr-xr-x 1 root root  408096 Nov 18 03:19 slurmctld_nonstop.so
-rwxr-xr-x 1 root root   65736 Nov 18 03:19 spank_pbs.so
drwxr-xr-x 4 root root      31 Nov 18 03:19 src
-rwxr-xr-x 1 root root  283944 Nov 18 03:19 switch_cray_aries.so
-rwxr-xr-x 1 root root  260496 Nov 18 03:19 switch_generic.so
-rwxr-xr-x 1 root root  209224 Nov 18 03:19 switch_none.so
-rwxr-xr-x 1 root root  430760 Nov 18 03:19 task_affinity.so
-rwxr-xr-x 1 root root  425856 Nov 18 03:19 task_cgroup.so
-rwxr-xr-x 1 root root  224216 Nov 18 03:19 task_cray_aries.so
-rwxr-xr-x 1 root root  205232 Nov 18 03:19 task_none.so
-rwxr-xr-x 1 root root  196728 Nov 18 03:19 topology_3d_torus.so
-rwxr-xr-x 1 root root  228112 Nov 18 03:19 topology_hypercube.so
-rwxr-xr-x 1 root root  172776 Nov 18 03:19 topology_node_rank.so
-rwxr-xr-x 1 root root  148632 Nov 18 03:19 topology_none.so
-rwxr-xr-x 1 root root  202040 Nov 18 03:19 topology_tree.so
```

**Comment by research-computing-facility** [comment]
Correction it does but it just is not linking, why??
```
root@papr-res-compute40:/galaxy/slurm-drmaa/slurm-drmaa-1.1.0#
==> ls -l /usr/lib64/libslurm.so
lrwxrwxrwx 1 root root 18 Nov 18 03:38 /usr/lib64/libslurm.so -> libslurm.so.34.0.0

```
Have tried with and without setting lib folder 
`./configure --prefix /galaxy/slurm-drmaa/compiled --with-slurm-inc=/usr/include --with-slurm-lib=/usr/lib64`

**Comment by research-computing-facility** [comment]
Hi @dpryan79 you were right I did a silly mistake I removed the whole line in the configure rather than leaving
`SLURM_LIBS="-lslurm "`
thanks for your insight!

**Comment by natefoo** [comment]
Fixed by @EricR86 in #34 and released in version [1.1.1](https://github.com/natefoo/slurm-drmaa/releases/tag/1.1.1). Thanks!

**Comment by X-WJ** [comment]
Hi, I also met this problem  as natefoo described .and i revised this line in configure  like this .

 #line 14022       SLURM_LIBS="-lslurm"
